File: 백준/Silver/1926. 그림/그림.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n");

let count = 0;
let MAX = 0;
const [N, M] = input[0].split(" ").map(Number);
let matrix = [];

let dx = [-1, 1, 0, 0];
let dy = [0, 0, -1, 1];

for (let i = 1; i <= N; i++) {
  let line = input[i].split(" ").map(Number);
  matrix.push(line);
}

function bfs(x, y) {
  let queue = [[x, y]];
  let area = 1;
  let head = 0;
  
  matrix[x][y] = 0; // 시작점 방문 처리
  
  while (head < queue.length) {
    let [curX, curY] = queue[head++];
    
    for (let i = 0; i < 4; i++) {
      let nx = dx[i] + curX;
      let ny = dy[i] + curY;
      
      if (nx >= 0 && ny >= 0 && nx < N && ny < M && matrix[nx][ny] === 1) {
        queue.push([nx, ny]);
        matrix[nx][ny] = 0; // 방문 처리
        area++;
      }
    }
  }
  
  return area;
}

for (let i = 0; i < N; i++) {
  for (let j = 0; j < M; j++) {
    if (matrix[i][j] === 1) {
      const result = bfs(i, j);
      count++;
      MAX = Math.max(MAX, result);
    }
  }
}

console.log(count);
console.log(MAX);

File: 백준/Silver/1926. 그림/그림.py
from collections import deque

N, M = map(int, input().split())

matrix = []
dx = [-1, 1, 0, 0]
dy = [0, 0, -1, 1] 

for _ in range(N):
    matrix.append(list(map(int, input().split()))) 

result = 0
MAX = 0

def bfs(start_x, start_y):
    q = deque()
    q.append([start_x, start_y])
    matrix[start_x][start_y] = 0  # 방문 처리
    
    area = 1  # 현재 그림의 넓이
    
    while q:
        curX, curY = q.popleft()
        
        for i in range(4):
            nx = curX + dx[i]
            ny = curY + dy[i]
            
            if nx >= 0 and ny >= 0 and nx < N and ny < M and matrix[nx][ny] == 1:
                q.append([nx, ny])
                matrix[nx][ny] = 0  # 방문 처리
                area += 1  # 넓이 증가
    
    return area

for i in range(N):
    for j in range(M):
        if matrix[i][j] == 1:
            area = bfs(i, j)
            MAX = max(MAX, area)
            result += 1

print(str(result) + "\n" + str(MAX))